### ISSUE TITLE
Custom binary_crossentropy that penalizes one mispredicted class

### ISSUE THREAD

**Issue by vabatista** [fchollet/keras#3068]
Hi, I have a dataset with 95% of data within one class and 5% of another. My models are having 94% of accuracy, but the model predicts 100% the prevalent class. My best roc_auc_score is about 0.71 on validation set.

Can anyone suggest a good custom objetive function that solves this problem?

**Comment by ChristianThomae** [comment]
You may use `class_weight` with inverse class frequencies when calling `fit`, see the [docs](http://keras.io/models/sequential/#fit).

**Comment by Yingyingzhang15** [comment]
@ChristianThomae Can you give an example of how to use the class-weight? I mean do I need to pass an array or something else to the `fit `function?

**Comment by tboquet** [comment]
You could also take a look at #2115 if you want an example of a loss function where you can weight the different cases of a confusion matrix.

**Comment by Yingyingzhang15** [comment]
@tboquet Thank you! It seems that I have solved this problem.The class weight should be a dictionary as explained in docs.
